editor/editor_file_system.cpp

```diff
@@ -463,12 +483,33 @@ void EditorFileSystem::_thread_func(void *_userdata) {
 	sd->_scan_filesystem();
 }
 
-bool EditorFileSystem::_test_for_reimport(const String &p_path, bool p_only_imported_files) {
-	if (!reimport_on_missing_imported_files && p_only_imported_files) {
-		return false;
+bool EditorFileSystem::_is_test_for_reimport_needed(const String &p_path, uint64_t p_last_modification_time, uint64_t p_modification_time, uint64_t p_last_import_modification_time, uint64_t p_import_modification_time, const Vector<String> &p_import_dest_paths) {
+	// The idea here is to trust the cache if the last modification times of the file and the .import file
+	// are the same. That means the files did not change since the last importation and that the files
```

Comment:
This sounds like it checks whether modification time of file is the same as modification time of .import. Needs slight rewording.

Reply:
Indeed, it was not clear at all, I think it's better now.